### PR TITLE
[ML] add multi node integ tests for data frames

### PR DESCRIPTION
**Pull request by benwtrent**
This PR adds the ground work for native, multi-node, integration tests for data frames. 

These tests brought up some interesting serialization bugs, most of them were addressed while this work was in flight, but some more were found once the latest changes were merged in.

### CODE REVIEW

File: x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/action/TransportGetDataFrameTransformsStatsAction.java [hunk 1]
```diff
@@ -172,10 +158,16 @@ private void collectStatsForTransformsWithoutTasks(Request request,
...
+                    for(ShardSearchFailure shardSearchFailure : searchResponse.getShardFailures()) {
+                        String nodeId = "";
+                        if (shardSearchFailure.shard() != null) {
+                            nodeId = shardSearchFailure.shard().getNodeId();
+                        }
+                        nodeFailures.add(new FailedNodeException(nodeId, shardSearchFailure.toString(), shardSearchFailure.getCause()));
+                    }
                     String msg = "transform statistics document search returned shard failures: " +
                         Arrays.toString(searchResponse.getShardFailures());
                     logger.error(msg);
```
Comment (hendrikmuhs) [inline x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/action/TransportGetDataFrameTransformsStatsAction.java]: nit: msg could be collapsed into `logger.error` with formating after this change

**Review by hendrikmuhs — APPROVED** [review]
LGTM

**Comment by droberts195** [comment]
I don't think these need to be "native", as they don't use any C++ functionality.  So I would rename `native-multi-node-tests` to just `multi-node-tests`.

**Review by davidkyle — APPROVED** [review]
LGTM

File: x-pack/plugin/data-frame/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/dataframe/integration/DataFrameIntegTestCase.java [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+            assertEquals(checkpoint, getDataFrameTransformStats(id)
+                .getTransformsStateAndStats()
+                .get(0)
+                .getTransformState()
+                .getCheckpoint()),
+            waitTime.getMillis(),
+            TimeUnit.MILLISECONDS);
+    }
+
+    protected void registerTransform(DataFrameTransformConfig config) {
```
[the remark this answers is not in the document]
Comment (benwtrent) [reply x-pack/plugin/data-frame/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/dataframe/integration/DataFrameIntegTestCase.java]: For sure, I was thinking as much myself initially :)

**Comment by benwtrent** [comment]
run elasticsearch-ci/1

**Pull request merged**